### ISSUE TITLE
Throw more informative error message if CmaEsSampler is used with categorical variables

### ISSUE THREAD

**Issue by Chillee** [optuna/optuna#1656]
Currently, if CmaEsSampler is used with categorical variables, it throws this error
```
The parameter 'x' in trial#1 is sampled independently by using `RandomSampler` instead of `CmaEsSampler` (optimization performance may be degraded). You can suppress this warning by setting `warn_independent_sampling` to `False` in the constructor of `CmaEsSampler`, if this independent sampling is intended behavior.
```
As somebody who isn't familiar with the details of CmaEsSampler, I was not aware that it didn't work with categorical variables, and thought it was an issue with how I was setting the parameter ranges

If the CmaEsSampler is called with categorical variables, it should probably throw an error/warning upfront explaining the issue.

**Comment by toshihikoyanase** [comment]
`CmaEsSampler` excludes categorical variables [here](https://github.com/optuna/optuna/blob/master/optuna/samplers/_cmaes.py#L158-L169). IMO, we can include the cause into the warning message, so that users can fix their code.

The log is generated in the `sample_independent` method ([here](https://github.com/optuna/optuna/blob/master/optuna/samplers/_cmaes.py#L299)), and we can, for example, modify it to show the following message:

> [W 2020-08-21 11:55:30,608] The parameter 'y' in trial#19 is sampled independently by using `RandomSampler` instead of `CmaEsSampler` **because it does not support `CategoricalDistribution`**(optimization performance may be degraded). You can suppress this warning by setting `warn_independent_sampling` to `False` in the constructor of `CmaEsSampler`, if this independent sampling is intended behavior.

I think this is good for `contribution-welcome`, so let me add the label.

**Comment by hvy** [comment]
Maybe we could mention `PyCmaSampler` as an alternative to just suppressing the warning, too.

**Comment by norihitoishida** [comment]
I'll work on this

**Comment by norihitoishida** [comment]
Hi @Chillee ! Thanks for your issue.
Could you show me your objective function please?

**Comment by norihitoishida** [comment]
Since `CmaEsSampler` already can exclude categorical distribution and send out info messages, It seems easily to solve this issue by modifing the message refer to @toshihikoyanase 's post.　

However some scenarios seem to confuse for users. 
In the case of the following objective function, the message looks like this:

**objective function**
```
def objective(trial):
    x = trial.suggest_uniform("x", -1, 1)
    y = trial.suggest_categorical("y", [-1,0,1])
    return x ** 2 + y ** 2
```
**info message**
```
 `CmaEsSampler` only supports two or more dimensional continuous search space. `RandomSampler` is used instead of `CmaEsSampler`.
 ```

https://colab.research.google.com/drive/1K7E8mXVrWrGDj07jTPPk00-uYNQLiVz_?usp=sharing

It would be difficult for users who aren't familiar with `CmaEsSampler` to understand from this message that CmaEsSampler ` doesn't support categorical distributions, so I think we should change the message in another part as well.

I'll continue to implement this.
If you have any ideas, please let me know :)

**Comment by c-bata** [comment]
Resolved at #2019. @norihitoishida Thank you for your contribution 👍
